cpp/src/arrow/compute/kernels/scalar_string_benchmark.cc

```diff
@@ -66,6 +66,11 @@ static void MatchSubstring(benchmark::State& state) {
   UnaryStringBenchmark(state, "match_substring", &options);
 }
 
+static void SplitPattern(benchmark::State& state) {
+  SplitPatternOptions options("abac");
+  UnaryStringBenchmark(state, "split_pattern", &options);
```

Comment:
Is there really a point in benchmarking random data? Presumably 0 split will happen...
You also don't need to worry about benchmarks in this PR, they may be added later.

Reply:
I've changed the pattern to "a", which is more meaningful for a true benchmark. But I use these as a sanity check if there aren't and major performance issues after refactoring, so I'd like to keep this in.